File: 雷霆象棋/start-server.py
#!/usr/bin/env python3
"""
雷霆中国象棋 - 快速启动服务器
使用方法：
    python start-server.py
    
或者指定端口：
    python start-server.py 8080
"""

import http.server
import socketserver
import sys
import os

# 默认端口
PORT = 8000

# 允许从命令行指定端口
if len(sys.argv) > 1:
    try:
        PORT = int(sys.argv[1])
    except ValueError:
        print(f"错误：无效的端口号 '{sys.argv[1]}'")
        print("用法：python start-server.py [端口]")
        sys.exit(1)

# 自定义请求处理，添加WASM的MIME类型
class CustomHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
    def end_headers(self):
        # 添加CORS头，允许跨域（开发时使用）
        self.send_header('Access-Control-Allow-Origin', '*')
        super().end_headers()
    
    def guess_type(self, path):
        # 确保.wasm文件使用正确的MIME类型
        if path.endswith('.wasm'):
            return 'application/wasm'
        return super().guess_type(path)

# 检查必要的文件
def check_files():
    required_files = [
        'index.html',
        'style.css',
        'script.js',
        'ai-engine.js',
        'opening-book.js',
        'ffish-worker.js'
    ]
    
    missing = []
    for file in required_files:
        if not os.path.exists(file):
            missing.append(file)
    
    if missing:
        print("⚠️  警告：以下文件缺失：")
        for file in missing:
            print(f"   - {file}")
        print()
    
    # 检查ffish文件
    if not os.path.exists('js/ffish.js'):
        print("❌ 错误：找不到 js/ffish.js")
        print("   请从npm或GitHub下载ffish.js并放入js文件夹")
        print("   查看README.md获取详细说明")
        print()
    
    if not os.path.exists('js/ffish.wasm'):
        print("❌ 错误：找不到 js/ffish.wasm")
        print("   请从npm或GitHub下载ffish.wasm并放入js文件夹")
        print("   查看README.md获取详细说明")
        print()
    
    if os.path.exists('js/ffish.js') and os.path.exists('js/ffish.wasm'):
        print("✅ ffish引擎文件已找到")
        return True
    else:
        print("⚠️  ffish引擎文件缺失，AI功能可能无法正常工作")
        print("   但你可以玩人人对战模式")
        return False

# 主函数
def main():
    print("=" * 50)
    print("   雷霆中国象棋 - 本地服务器")
    print("=" * 50)
    print()
    
    # 检查文件
    ffish_ok = check_files()
    print()
    
    # 创建服务器
    with socketserver.TCPServer(("", PORT), CustomHTTPRequestHandler) as httpd:
        print(f"🚀 服务器启动成功！")
        print(f"   访问地址：http://localhost:{PORT}")
        print()
        
        if ffish_ok:
            print("🎮 游戏已就绪，可以开始对战！")
        else:
            print("🎮 人人对战模式可用")
            print("🤖 人机对战需要ffish引擎文件")
        
        print()
        print("按 Ctrl+C 停止服务器")
        print("-" * 50)
        
        try:
            httpd.serve_forever()
        except KeyboardInterrupt:
            print()
            print()
            print("👋 服务器已停止")
            print("感谢使用雷霆中国象棋！")

if __name__ == "__main__":
    main()

File: start-server.js
#!/usr/bin/env node
/**
 * 雷霆中国象棋 - 快速启动服务器 (Node.js版本)
 * 使用方法：
 *     node start-server.js
 *     
 * 或者指定端口：
 *     node start-server.js 8080
 */

const http = require('http');
const fs = require('fs');
const path = require('path');

const PORT = process.argv[2] || 8000;

// MIME类型映射
const mimeTypes = {
    '.html': 'text/html',
    '.js': 'text/javascript',
    '.css': 'text/css',
    '.json': 'application/json',
    '.png': 'image/png',
    '.jpg': 'image/jpg',
    '.gif': 'image/gif',
    '.svg': 'image/svg+xml',
    '.wav': 'audio/wav',
    '.mp3': 'audio/mpeg',
    '.wasm': 'application/wasm',
    '.ico': 'image/x-icon'
};

// 检查文件是否存在
function checkFiles() {
    const requiredFiles = [
        'index.html',
        'style.css',
        'script.js',
        'ai-engine.js',
        'opening-book.js',
        'ffish-worker.js'
    ];
    
    const missing = requiredFiles.filter(file => !fs.existsSync(file));
    
    if (missing.length > 0) {
        console.log('⚠️  警告：以下文件缺失：');
        missing.forEach(file => console.log(`   - ${file}`));
        console.log();
    }
    
    // 检查ffish文件
    const ffishJsExists = fs.existsSync('js/ffish.js');
    const ffishWasmExists = fs.existsSync('js/ffish.wasm');
    
    if (!ffishJsExists) {
        console.log('❌ 错误：找不到 js/ffish.js');
        console.log('   请从npm或GitHub下载ffish.js并放入js文件夹');
        console.log('   查看README.md获取详细说明');
        console.log();
    }
    
    if (!ffishWasmExists) {
        console.log('❌ 错误：找不到 js/ffish.wasm');
        console.log('   请从npm或GitHub下载ffish.wasm并放入js文件夹');
        console.log('   查看README.md获取详细说明');
        console.log();
    }
    
    if (ffishJsExists && ffishWasmExists) {
        console.log('✅ ffish引擎文件已找到');
        return true;
    } else {
        console.log('⚠️  ffish引擎文件缺失，AI功能可能无法正常工作');
        console.log('   但你可以玩人人对战模式');
        return false;
    }
}

// 创建服务器
const server = http.createServer((req, res) => {
    console.log(`${new Date().toISOString()} - ${req.method} ${req.url}`);
    
    // 添加CORS头
    res.setHeader('Access-Control-Allow-Origin', '*');
    res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
    res.setHeader('Access-Control-Allow-Headers', 'Content-Type');
    
    if (req.method === 'OPTIONS') {
        res.writeHead(200);
        res.end();
        return;
    }
    
    let filePath = '.' + req.url;
    if (filePath === './') {
        filePath = './index.html';
    }
    
    const extname = String(path.extname(filePath)).toLowerCase();
    const contentType = mimeTypes[extname] || 'application/octet-stream';
    
    fs.readFile(filePath, (error, content) => {
        if (error) {
            if (error.code === 'ENOENT') {
                res.writeHead(404, { 'Content-Type': 'text/html' });
                res.end('<h1>404 Not Found</h1>', 'utf-8');
            } else {
                res.writeHead(500);
                res.end('Sorry, check with the site admin for error: ' + error.code + ' ..\n');
            }
        } else {
            res.writeHead(200, { 'Content-Type': contentType });
            res.end(content, 'utf-8');
        }
    });
});

// 启动服务器
console.log('='.repeat(50));
console.log('   雷霆中国象棋 - 本地服务器');
console.log('='.repeat(50));
console.log();

const ffishOk = checkFiles();
console.log();

server.listen(PORT, () => {
    console.log(`🚀 服务器启动成功！`);
    console.log(`   访问地址：http://localhost:${PORT}`);
    console.log();
    
    if (ffishOk) {
        console.log('🎮 游戏已就绪，可以开始对战！');
    } else {
        console.log('🎮 人人对战模式可用');
        console.log('🤖 人机对战需要ffish引擎文件');
    }
    
    console.log();
    console.log('按 Ctrl+C 停止服务器');
    console.log('-'.repeat(50));
});

// 处理退出
process.on('SIGINT', () => {
    console.log();
    console.log();
    console.log('👋 服务器已停止');
    console.log('感谢使用雷霆中国象棋！');
    process.exit(0);
});
